cpp/arcticdb/entity/metrics.hpp

```diff
@@ -34,6 +35,41 @@ const std::string PROMETHEUS_ENV_LABEL = "env";
 const int SUMMARY_MAX_AGE = 30;
 const int SUMMARY_AGE_BUCKETS = 5;
 
+class PrometheusConfig {
+public:
+    enum Model {
+        NO_INIT = 0,
+        PUSH = 1,
+        WEB = 2
+    };
+    PrometheusConfig() : model(Model::NO_INIT) {}
+
+    PrometheusConfig(const std::string& host,
+                     const std::string& port,
+                     const std::string& job_name, 
+                     const std::string& instance, 
+                     const std::string& prometheus_env, 
+                     const int model)
+            : host(host)
+            , port(port)
+            , job_name(job_name)
+            , instance(instance)
+            , prometheus_env(prometheus_env)
+            , model(static_cast<Model>(model)) {
+                util::check(!host.empty(), "PrometheusConfig: host is empty");
+                util::check(!port.empty(), "PrometheusConfig: port is empty");
+                util::check(!job_name.empty(), "PrometheusConfig: job_name is empty");
+                util::check(!instance.empty(), "PrometheusConfig: instance is empty");
+                util::check(!prometheus_env.empty(), "PrometheusConfig: prometheus_env is empty");
+            }
+    
+    std::string host;
+    std::string port;
+    std::string job_name;
+    std::string instance;
+    std::string prometheus_env;
+    Model model;
```

Comment:
(nit) We usually have a trailing underscore on the data members names, eg `model_`